cmd/amp/cli/cli_test.go

```diff
@@ -138,8 +137,8 @@ func runTestSpec(t *testing.T, test *TestSpec) (err error) {
 				t.Log(err)
 			}
 
-			endTime := time.Now().UnixNano() / 1000000
-			if cmdSpec.Timeout != 0 && endTime-startTime >= cmdSpec.Timeout {
+			endTime := time.Since(startTime)
+			if cmdSpec.Timeout != 0 && int64(endTime/time.Millisecond) >= cmdSpec.Timeout {
```

Comment:
See this [example](https://gobyexample.com/timeouts) on how to use a channel for executing code after a timeout to kill the process and synchronize on the success or failure path before returning.